### DISCUSSION TITLE
Nested Object Validation

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#5086] [Feature Requests]
Directus' validation only supports "simple" object structures, like

```json
{
  "field": {
    "_eq": "value"
  }
}
```

we should extend this with the same support for nesting as filters and permissions, so you can do things like

```json
{
  "author": {
    "first_name": {
      "_eq": "Rijk"
    }
  }
}
```

to validate that any new author that's created within this parent item has to have the name `Rijk`

**Comment by setbusiness** [comment]
yes for example filter by user who have a relationship with a specific company data model with dynamic values...but how can we create or access at of the list of dyamic values ? thx

**Comment by benhaynes** [comment]
Ref: https://github.com/directus/directus/discussions/2993
Ref: https://github.com/directus/directus/discussions/3914

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by Lukas-Sachse** [reply]
@rijkvanzanten 5 Upvotes + 11 Thumbs up could be combined to 16 "positive votes".

I discovered, a moment ago, that I would need this. 


### My use case:
#### Filtered Selection ✅ 
1. M2O: Country
2. M2O: Federal_state → selection (interface filter) based on `federal_state.country.id Equals {{ country }}`.
3. M2O: District → selection (interface filter) based on `district.federal_state.id Equals {{ federal_state }}`.

A user can only choose a federal_state (ex. Saxony) if it belongs to the previous selected country (ex. Germany).
Germany → Saxony → Leipzig ✅

#### Filtered Validation ❌
But after a first full selection process, a user can change the country to whatever he wants. (ex. Switzerland)
Switzerland → Saxony → Leipzig ❌

As for now, there is no way to prevent this invalid combination with the build in validation.
